js/show-graph.js

```diff
@@ -184,24 +152,67 @@ diagram.showGraph = function(args) {
-        if (d3.event.defaultPrevented) return
-        container.selectAll("."+d.startOf)
-          .style('stroke', '#'+colorpicker.value)
-          .style('fill', function(d) { return d.bobbin ? '#'+colorpicker.value : 'none' })
-    })
-
+//    var colorpicker = (args.threadColor == undefined ? undefined : d3.select(args.threadColor)[0][0])
+//    if (colorpicker) {
+//        threadStarts.on('click', function (d) {
+//            if (d3.event.defaultPrevented) return
+//            container.selectAll("."+d.startOf)
+//              .style('stroke', '#'+colorpicker.value)
+//              .style('fill', function(d) { return d.bobbin ? '#'+colorpicker.value : 'none' })
+//        })
+//    }
+
+    var drawPath = function(d) {
+        var source = args.nodes[d.source]
+        var target = args.nodes[d.target]
+        var sX = source.x
+        var sY = source.y
+        var tX = target.x
+        var tY = target.y
+        var dX = (tX - sX)
+        var dY = (tY - sY)
+        var mid = d.left ? ("S" + (sX - dY/3 + dX/3) + "," + (sY + dX/3 + dY/3)) :
+                  d.right ? ("S" + (sX + dY/3 + dX/3) + "," + (sY + dY/3 - dX/3)) :
+                             " "
+        if (d.end && d.end == "white")
+            return "M"+ sX + "," + sY + mid + " " + (tX - dX/4) + "," + (tY - dY/4)
+        if (d.start && d.start == "white")
+            return "M"+ (sX + dX/4) + "," + (sY + dY/4) + mid + " " + tX + "," + tY
+        if (d.mid)
+            return "M"+ sX + "," + sY + " " + (sX + dX/2) + "," + (sY + dY/2) + " " + tX + "," + tY
+        return "M"+ sX + "," + sY + " " + tX + "," + tY
+    }
+    var moveNode = function(d) {
+        return "translate(" + d.x + "," + d.y + ")"
+    }
     // a higher speed for IE as marks only appear when the animation is finished
     var mod = isMobileMac ? 10 : isIE ? 3 : 2
     var step = 0
-    // layout simulation step
-    force.on("tick", function() {
-        if ( ((step++)%mod) != 0) return
-        link.attr("d", diagram.path)
-        node.attr("transform", diagram.transform)
-    })
-    force.on("end", function(){
-      if (isIE) diagram.markLinks(link)
-      if (args.onAnimationEnd) args.onAnimationEnd()
-    })
+    var simTicked = function() {
+                         if ( ((step++)%mod) != 0) return // skip rendering
+                         nodes.attr("transform", moveNode)
+                         links.attr("d", drawPath)
+                     }
+    var simEnded = function(){
+                      if (isIE) diagram.markLinks(links)
+                      if (args.onAnimationEnd) args.onAnimationEnd()
+                    }
+//    var force = d3.layout.force()
+//        .nodes(args.nodes)
+//        .links(args.links)
+//        .size([args.width, args.height])
+//        .charge(-20)
+//        .linkDistance(10)
+//        .linkStrength(10)
+//        .start()
+//        .alpha(0.01)
+// Above calibration with the v3 API resulted in a relative quick compact animation.
```

Comment:
The comment above shows the [current calibration](https://github.com/d-bl/GroundForge/55ed13e/gh-pages/js/show-graph.js#L165-L173) of the diagrams in the gh-pages branch with v3 of D3js. Some older versions:
* https://github.com/d-bl/GroundForge/blob/fd295cfbc/js/show-graph.js#L127-L135
* https://github.com/d-bl/GroundForge/blob/29ac9b0ba/show-graph.js#L57-L64
* https://github.com/d-bl/GroundForge/blob/84eee36/show-graph.js#L30-L34

You could paste the values into the gh-pages branch to see their effects on the diagrams.